apiserver/facades/agent/upgrader/register.go

```diff
@@ -31,7 +33,13 @@ func Register(registry facade.FacadeRegistry) {
 // newUpgraderFacade provides the signature required for facade registration.
 func newUpgraderFacade(ctx facade.ModelContext) (Upgrader, error) {
 	auth := ctx.Auth()
-	st := ctx.State()
+
+	if !auth.AuthMachineAgent() &&
+		!auth.AuthModelAgent() &&
+		!auth.AuthUnitAgent() {
```

Comment:
This was copied, but seems wrong.
We want one of controller tag, machine tag, unit tag.
Currently, sadly, `AuthMachineAgent()` works for both machine and controller tags.
So we can drop the `AuthModelAgent()` right?